compiler/noirc_frontend/src/resolve_locations.rs

```diff
@@ -167,4 +168,21 @@ impl NodeInterner {
                 method.map(|method| method.location)
             })
     }
+
+    #[tracing::instrument(skip(self), ret)]
+    fn try_resolve_type_alias(&self, location: Location) -> Option<Location> {
+        self.type_alias_ref
+            .iter()
+            .find(|(type_alias_id, named_type_location)| {
```

Comment:
Iterating through each type alias reference and testing the location will be slow in larger programs. We do something similarly with searching through other locations currently.

In the future I think we should use rangemaps instead - possibly in a separate struct from the NodeInterner since I'd like for more separation of concerns in the future.

Reply:
agreed